### PR TITLE
drivers/servo: Add test program, handle inexact frequencies

### PR DESCRIPTION
**Pull request by gebart**
This improves the initialization procedure of the servo driver to account for limited frequency selections in the PWM module of some controllers (e.g. Kinetis)

Added a simple test program to try it out on a real servo. Connect an RC servo to `PWM_0` channel 0 to see any results.

### CODE REVIEW

File: drivers/servo/servo.c [hunk 1]
```diff
@@ -21,26 +23,80 @@
 #include "servo.h"
 #include "periph/pwm.h"
 
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+#define SEC_IN_USEC     (1000000U)
```
Comment (PeterKietzmann) [inline drivers/servo/servo.c]: This define already exists in timex.h

**Comment by PeterKietzmann** [comment]
PR looks good to me, I like this improvement! This is Haukes "baby" so he should also have a look at it.

File: drivers/servo/servo.c (hunk 1 above)
Comment (gebart) [inline drivers/servo/servo.c]: I found it when I `git grep`-ed during writing, but decided that the test shouldn't be dependent on timex.h just to replace a simple literal. OK to leave this in?

Comment (PeterKietzmann) [inline drivers/servo/servo.c]: I don't really mind. Just remembered that I've seen this "issue" commented in some PRs. Let's see what Hauke says.

Comment (OlegHahm) [inline drivers/servo/servo.c]: Including `timex.h` isn't pulling in any module dependency.

Comment (gebart) [inline drivers/servo/servo.c]: @OlegHahm because timex is part of `/sys/include`?

Comment (haukepetersen) [inline drivers/servo/servo.c]: yepp.

**Comment by gebart** [comment]
changed to use timex.h for SEC_IN_USEC

**Comment by haukepetersen** [comment]
looks good to me and it is very useful. Will test this weekend!

**Comment by gebart** [comment]
OK to squash?

**Comment by PeterKietzmann** [comment]
Yes, please squash. I think Hauke just wants to test the PR and will merge then. ACK from my side, for the code.

**Comment by gebart** [comment]
rebased, squashed

**Comment by haukepetersen** [comment]
just working on the PWM driver for the samr21 board, Will test this PR as soon as I have the PWM working...

**Comment by haukepetersen** [comment]
works as expected, Travis is happy -> ACK and go.

**Pull request merged**